File: src/lib/api/codexLocale.ts
import { invoke } from "@tauri-apps/api/core";

export interface CodexLocaleStatus {
  installed: boolean;
  version: string | null;
  chineseResourcesAvailable: boolean;
  localeOverride: string | null;
  chineseEnabled: boolean;
  restartRequired: boolean;
}

export const codexLocaleApi = {
  getStatus(): Promise<CodexLocaleStatus> {
    return invoke("get_codex_locale_status");
  },

  setSimplifiedChinese(enabled: boolean): Promise<CodexLocaleStatus> {
    return invoke("set_codex_simplified_chinese", { enabled });
  },
};

File: src-tauri/src/commands/codex_locale.rs
use crate::codex_locale::{self, CodexLocaleStatus};

#[tauri::command]
pub async fn get_codex_locale_status() -> Result<CodexLocaleStatus, String> {
    codex_locale::get_status().map_err(Into::into)
}

#[tauri::command]
pub async fn set_codex_simplified_chinese(enabled: bool) -> Result<CodexLocaleStatus, String> {
    codex_locale::set_simplified_chinese(enabled).map_err(Into::into)
}
